packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/snapshot_list.dart

```diff
@@ -120,11 +120,16 @@ class _SnapshotListItemsState extends State<_SnapshotListItems>
   void initState() {
     super.initState();
     _scrollController = ScrollController();
+    _initWidget();
   }
 
   @override
-  void didChangeDependencies() {
-    super.didChangeDependencies();
+  void didUpdateWidget(covariant _SnapshotListItems oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.controller != widget.controller) _initWidget();
+  }
+
+  void _initWidget() {
```

Comment:
nit: rename to initListener since that is what this method is doing

Reply:
This method is intended to do whatever is needed to init widget. It is like 'initState'. I would leave this name and have it consistent across the code: _initWidget is a method that is invoked in initState and didUpdateWidget.
What do you think?

Reply:
The naming is a bit misleading. We are not initializing the widget. The widget has already been created and passed to the state, and widgets are immutable so we will never do any "initialization" for the widget. What we are doing is setting up the state for the widget. I'd either change to 'init' or 'initListener' to better reflect what this method is doing.